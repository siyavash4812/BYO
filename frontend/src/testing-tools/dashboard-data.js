export const DashboardData = [
  {
    id: 1,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "AAA",
  },
  {
    id: 2,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "BBB",
  },
  {
    id: 3,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "B2B",
  },
  {
    id: 4,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "B2C",
  },
  {
    id: 5,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "22B",
  },
  {
    id: 6,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "133",
  },
  {
    id: 7,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "TDD",
  },
  {
    id: 8,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "BFD",
  },
  {
    id: 9,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "1FF",
  },
  {
    id: 10,
    createdAt: "2021-08-11T13:38:49.000Z",
    updatedAt: "2021-08-11T13:38:49.000Z",
    title: "CCC",
  },
];
