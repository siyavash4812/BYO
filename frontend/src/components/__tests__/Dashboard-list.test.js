import React from "react";
import renderer from "react-test-renderer";
import { DashboardData } from "../../testing-tools/dashboard-data";
import DashboardList from "../Dashboard-list";

describe("DashboardList", () => {
  it("renders DashboardList", () => {
    const tree = renderer
      .create(<DashboardList data={DashboardData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
