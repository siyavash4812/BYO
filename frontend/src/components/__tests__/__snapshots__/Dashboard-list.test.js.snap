// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardList renders DashboardList 1`] = `
.c2 {
  font-size: 1em;
  color: #001f3f;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  border: 1px dotted peru;
  margin: 10px 0px;
  background-color: #DDDDDD;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
  data-testid="dashboard-list"
>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      AAA
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      BBB
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      B2B
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      B2C
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      22B
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      133
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      TDD
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      BFD
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      1FF
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
  <div
    className="c1"
    data-testid="dashboard-card"
  >
    <span
      className="c2"
      size="S"
    >
      Title: 
      CCC
    </span>
    <span
      className="c2"
      size="S"
    >
      Created: 
      2021-08-11 14-38-49
    </span>
  </div>
</div>
`;
